Snakefile

```diff
@@ -0,0 +1,267 @@
+    '''
+
+##########################################################
+## Default sourmash plots
+##########################################################
+
+rule sourmash_plot:
+   input: "outputs/sourmash_compare/comp_k{ksize}" 
+   output: "outputs/sourmash_plot/comp_k{ksize}.hist.pdf"
+   params: outdir = "outputs/sourmash_plot"
+   conda: "envs/sourmash.yml"
+   shell:'''
+   sourmash plot --pdf --labels --output-dir {params.outdir} 
+   '''
+
+##########################################################
+## Sourmash gather
+##########################################################
+
+rule sourmash_gather:
+    input:
+        sig="outputs/sourmash_sketch/{samples}ass.sig",
+        databases=expand("inputs/sourmash_databases/genbank-2022.03-{lineage}-k{{ksize}}.zip", lineage = LINEAGES)
+    output: csv="outputs/sourmash_gather/{samples}ass-vs-genbank-2022.03-k{ksize}.csv"
+    conda: "envs/sourmash.yml"
+    shell:'''
+    sourmash gather -k {wildcards.ksize} --scaled 1000 --threshold-bp 0 -o {output.csv} {input.sig} {input.databases}
+    '''
+   
+##########################################################
+## Sourmash taxonomy
+##########################################################
+
+rule gunzip_lineage_csvs:
+    input: "inputs/sourmash_databases/genbank-2022.03-{lineage}.lineages.csv.gz"
+    output: "inputs/sourmash_databases/genbank-2022.03-{lineage}.lineages.csv"
+    shell:'''
+    gunzip -c {input} > {output}
+    '''
+
+rule sourmash_taxonomy_prepare:
+    input: expand("inputs/sourmash_databases/genbank-2022.03-{lineage}.lineages.csv", lineage = LINEAGES),
+    output: "outputs/sourmash_taxonomy/genbank-2022.03-prepared-lineages.sqldb"
+    conda: "envs/sourmash.yml"
+    shell:'''
+    sourmash tax prepare --taxonomy-csv {input} -o {output}
+    '''
+
+rule sourmash_taxonomy_annotate:
+   input:
+       lin_prepared="outputs/sourmash_taxonomy/genbank-2022.03-prepared-lineages.sqldb",
+       gather="outputs/sourmash_gather/{samples}ass-vs-genbank-2022.03-k{ksize}.csv"
+   output: "outputs/sourmash_taxonomy/{samples}ass-vs-genbank-2022.03-k{ksize}.with-lineages.csv"
+   params: outdir = "outputs/sourmash_taxonomy/"
+   conda: "envs/sourmash.yml"
+   shell:'''
+   sourmash tax annotate -g {input.gather} -t {input.lin_prepared} -o {params.outdir}
+   '''
+
+# visualizations:
+# upset plot on hashes
+# upset plot on taxonomy?
+# taxonomy bar charts as fraction of sample
+# ...
+
+####################################################################
+## Visualization: upset plot for the intersection of shared 
+##                hashes (subsampled k-mers) between samples
+####################################################################
+
+rule sourmash_sketch_convert_to_csv:
+    input: "outputs/sourmash_sketch/{samples}ass.sig"
+    output: "outputs/sourmash_sketch_csv/{samples}_k{ksize}.csv"
+    conda: "envs/sourmash.yml"
+    shell:'''
+    python scripts/sig_to_csv.py {wildcards.ksize} {input} {output}
+    '''
+
```

Comment:
nit: this is more of an aesthetic thing, but I like the consistency of it. Using a linter (or a text editor plugin), we should make sure there is only a single trailing newline at EOF